presto-main-base/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -1745,6 +1763,17 @@ private static Double array2Norm(Block array)
         return Math.sqrt(norm);
     }
 
+    private static boolean arraysHaveNull(Block leftArray, Block rightArray)
+    {
+        boolean hasNull = false;
+        if (leftArray.mayHaveNull() || rightArray.mayHaveNull()) {
+            for (int i = 0; i < leftArray.getPositionCount(); i++) {
+                hasNull = hasNull | leftArray.isNull(i) | rightArray.isNull(i);
+            }
+        }
+        return hasNull;
+    }
```

Comment:
Since this is not a common case (and looks like we want strict non-null guarantee), just doing the mayHaveNull cheks is enough - make it a checkArg(left.mayHvenull || right.mayHavenull) in the cller  so no performance overhead